docs/src/main/asciidoc/writing-extensions.adoc

```diff
@@ -538,6 +541,42 @@ So you may consider omitting explicit `name` in some cases.
 // The following link should point to the mojo page once https://github.com/quarkusio/quarkusio.github.io/issues/265 is fixed
 Please refer to https://github.com/quarkusio/quarkus/blob/{quarkus-version}/devtools/maven/src/main/java/io/quarkus/maven/CreateExtensionMojo.java[CreateExtensionMojo JavaDoc] for all the available options of the mojo.
 
+==== Using Gradle
+
+You will need to apply the `io.quarkus.extension` plugin in your `runtime` module and declare `quarkus-extension-processor` as annotation processor:
+The plugin adds a `extensionDescriptor` task that will generate the `quarkus-extension.properties` file.
+
+[source,groovy]
+----
+plugins {
+    id 'java'
+    id 'io.quarkus.extensions'
+}
+
+dependencies {
+    annotationProcessor 'io.quarkus:quarkus-extension-processor'
+}
+----
+
+In your deployment module, you need to add `quarkus-extension-processor` as annotation processor:
+
+[source,groovy]
+----
+plugins {
+    id 'java'
+}
+
+dependencies {
+    annotationProcessor 'io.quarkus:quarkus-extension-processor'
+}
+----
+
+[WARNING]
+====
+This plugin is still experimental. It will generate the `quarkus-extension.properties` file but not the `quarkus-extension.yml` file.
```

Comment:
```suggestion
This plugin is still experimental. It will generate the `META-INF/quarkus-extension.properties` file but not the `META-INF/quarkus-extension.yml` file.
```